### DISCUSSION TITLE
Has highlighting changed in neovim 0.9 or LSP or Treesitter?

### DISCUSSION THREAD

**Discussion by shauryagoel** [LazyVim/LazyVim#1950] [Q&A]
I have been using the LazyVim config since I made it in March 2023. I decided to recently update my neovim config to use neovim 0.9.4 and the latest version of LazyVim. However, there are slight differences in highlighting, such as unused variables in Python code are of dark colour, and in Lua, the `@param` is highlighted like normal code instead of being highlighted like a comment. Is this behaviour from the treesitter or LSP? Is this specific to 0.9? How can I remove this behaviour?

UPDATE: adding `client.server_capabilities.semanticTokensProvider = nil` in LspAttach autocmd has solved the problem for lua (don't know why). However, the Python problem of undefined variables remains.

**Comment by amaanq** [comment]
It's LSP, :h lsp-semantic_tokens

**Reply by shauryagoel** [reply]
Can you describe how I can use this information? It has 5 functions, where and how should I call these functions?

**Comment by dpetka2001** [comment]
You can use the command `:Inspect` or keymap `<leader>ui` to show you information on the position where your cursor is. If you don't like the behavior, maybe you could change the corresponding highlight-groups to something more aesthetically appealing to you.

**Reply by shauryagoel** [reply]
Okay, but how to do it for the undefined variables? I am only getting `@variable` in the `:Inspect`.
